scripts/st2bootstrap-deb.template.sh

```diff
@@ -104,6 +104,34 @@ setup_args() {
         exit 1
     fi
   fi
+
+  # Python 3.6 package is not available in Ubuntu Xenial
+  # Installer can add it via 3rd party PPA based on user agreement
+  if [[ "$SUBTYPE" = 'xenial' ]]; then
+    sudo apt-get update > /dev/null 2>/dev/null
+    # check if python3.6 is available
+    if (! apt-cache show python3.6 2> /dev/null | grep 'Package:' > /dev/null); then
+      echo ""
+      echo "WARNING!"
+      echo "python3.6 package is required as a dependency for StackStorm and none of your software repositories provide it."
+      echo "We recommend switching to Ubuntu 18.04 LTS (Bionic) as a base OS."
+      echo ""
+      echo "Alternatively we'll try to add 3rd party python3.6 APT repository: https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa."
+      echo "By continuing you're aware of security risk associated with using unofficial 3rd party PPA repository."
+      echo ""
+      read -p "Press [y] to continue or [n] to cancel adding it: " choice
+      case "$choice" in
+        y|Y )
+          sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F23C5A6CF475977595C89F51BA6932366A755776
+          echo "deb http://ppa.launchpad.net/deadsnakes/ppa/ubuntu xenial main" | sudo tee /etc/apt/sources.list.d/deadsnakes-ubuntu-ppa-xenial.list
```

Comment:
Instead of doing this manually, I would use the `add-apt-repository` command, which will take care of this for you:

```bash
sudo apt-get install --yes software-properties-common
sudo add-apt-repository --yes ppa:deadsnakes/ppa
sudo apt-get update
```

That should reduce our maintenance burden if the PPA changes at all.

Reply:
I tried that, but installing `software-properties-common` is not what I wanted in the foremost pre-req place of the installer script. See `setup_args` for more context.

I decided to go with adding the string manually as we already do that in other places.
I don't expect us to ever change this repository to another, but rather remove it alltogether in a half of the year.

Reply:
It's a big enough check that we could probably put it in its own Bash function/step.